randomwallpaper@iflow.space/wallpaperController.js

```diff
@@ -156,6 +157,9 @@ var WallpaperController = class {
 				case 3:
 					imageSourceAdapter = new GenericJsonAdapter.GenericJsonAdapter(source.id, this.wallpaperlocation);
 					break;
+				case 4:
+					imageSourceAdapter = new LocalFolderAdapter.LocalFolderAdapter(source.id, this.wallpaperlocation);
```

Comment:
The constructor crawls the selected folder recursively. This might be an issue for large folders. E.g., someone has a huge wallpaper collection and suddenly all images are investigated for their file type.
I have no idea of the overhead of `_listDirectory` right now but I'll create an issue for tracking this topic.

Reply:
For reference: https://github.com/ifl0w/RandomWallpaperGnome3/issues/162